### PR TITLE
ECDC-2428: Generate script from table in LoKI gui

### PR DESCRIPTION
**Pull request by ebadkamil**
This PR generates a semi-realistic script from the table in prototype panel, and sends it to the scripting panel editor.

### CODE REVIEW

File: nicos_ess/loki/gui/loki_scriptbuilder.py [hunk 1]
```diff
@@ -254,6 +254,49 @@ def on_clearTableButton_clicked(self):
...
+                    f"{self.comboSansDurationType.currentText()})\n"
+                )
+                # What to do if order is Simultaneous?
+                if self.comboTransOrder.currentText() == "TRANS First":
+                    count = do_trans + do_sans
+                else:
+                    count = do_sans + do_trans
+
+                template += (
+                    f"{filler['pre-command']}\n"
```
Comment (ebadkamil) [inline nicos_ess/loki/gui/loki_scriptbuilder.py:291]: Will it be a user-command like `move(something, ...)` etc?

**Comment by ebadkamil** [comment]
`tableScript` must  also have validators for columns.

File: nicos_ess/loki/gui/loki_scriptbuilder.py [hunk 2]
```diff
@@ -254,6 +265,50 @@ def on_clearTableButton_clicked(self):
...
+                    f"do_trans({filler['trans_duration']}, "
+                    f"{self.comboTransDurationType.currentText()})\n"
+                )
+                do_sans = (
+                    f"do_sans({filler['sans_duration']}, "
+                    f"{self.comboSansDurationType.currentText()})\n"
+                )
+                # What to do if order is Simultaneous?
+                if self._available_trans_options[
+                    self.comboTransOrder.currentText()] == TransOrder.TRANSFIRST:
```
Comment (ebadkamil) [inline nicos_ess/loki/gui/loki_scriptbuilder.py:297]: Better way to check order? Probably `TransOrder(IntEnum)` not required.

**Comment by mattclarke** [comment]
This is a good start.
We should get together and try and flesh out some of the details.

**Comment by ebadkamil** [comment]
@mattclarke alright, whenever you have time this week let me know.

**Comment by ebadkamil** [comment]
Relevant stuff from the inbox has been moved from TODO to DONE. Now ready for review.

**Review by mattclarke — APPROVED** [review]

**Comment by mattclarke** [comment]
There are a few follow-up items to do, but this can go in now as I don't want this to get any bigger.

**Pull request merged**